pynpoint/processing/psfpreparation.py

```diff
@@ -356,8 +359,6 @@ def run(self) -> None:
 
         index_sort = np.argsort(index)
 
-        nimages = self.m_image_in_port.get_shape()[0]
-
         frames = memory_frames(memory, nimages)
```

Comment:
Should this be adjusted for the 4D case, similar to the `PSFpreparationModule`?

Reply:
You are right. But this actually gives my an idea on how to optimize the frames problem with 4D data. I will investigated it and, if it works, upload it at a later time.

For now I changed the frames handling equivalent to the one used in the PsfPreparationModule.

Reply:
OK, thanks. The `i` index is already used within the for loop in which it is embedded (should have been picked up by `pylint`?). The `enumerate` can probably removed (see my previous commit).

Reply:
You are right. I left it because it was previously done this way. I now simplified the expression and corrected a minor bug.